lib/auth/auth_with_roles.go

```diff
@@ -2028,21 +2033,26 @@ func (a *ServerWithRoles) listResourcesWithSort(ctx context.Context, req proto.L
 		SearchKeywords: req.SearchKeywords,
 		StartKey:       req.StartKey,
 		EnrichResourceFn: func(r types.ResourceWithLabels) (types.ResourceWithLabels, error) {
-			if req.IncludeLogins && (r.GetKind() == types.KindNode || r.GetKind() == types.KindWindowsDesktop) {
-				resourceChecker, err := a.newResourceAccessChecker(req.ResourceType)
-				if err != nil {
-					return nil, trace.Wrap(err)
-				}
+			if !req.IncludeLogins && (r.GetKind() != types.KindNode || r.GetKind() != types.KindWindowsDesktop || r.GetKind() != types.KindAppServer) {
```

Comment:
nit: not sure how i feel about this check. Do we have to keep expanding this when new types is added to `GetAllowedLoginsForResource`? Could we maybe always call `GetAllowedLoginsForResource` for `req.IncludeLogins` and skip certain errors like `trace.NotImplemented`